lib/validate-schemas.spec.js

```diff
@@ -24,19 +24,157 @@ function loadRemoteSchema(uri) {
     });
 }
 
+const dataKeyTopicRegex = /^[a-zA-Z]([a-zA-Z0-9]+|\.|_)*[a-zA-Z0-9]$/;
+const dataKeyTopicMinLength = 0;
+const dataKeyTopicMaxLength = 64;
+const validatePropertyName = (errors, dataKey) => {
+  if (dataKey.length < dataKeyTopicMinLength) {
+    errors.push(`dataKey '${dataKey}' too short`);
+  }
+
+  if (dataKey.length > dataKeyTopicMaxLength) {
+    errors.push(`dataKey '${dataKey}' too long`);
+  }
+
+  if (!dataKeyTopicRegex.test(dataKey)) {
+    errors.push(`dataKey '${dataKey}' does not match convention`);
+  }
+};
+
+const validateTopic = (filepath, errors, topic) => {
+  if (topic === undefined) {
+    errors.push(`topic not set for ${filepath}`);
+    return;
+  }
+
+  if (topic.length < dataKeyTopicMinLength) {
+    errors.push(`topic '${topic}' too short`);
+  }
+
+  if (topic.length > dataKeyTopicMaxLength) {
+    errors.push(`topic '${topic}' too long`);
+  }
+
+  if (!dataKeyTopicRegex.test(topic)) {
+    errors.push(`topic '${topic}' does not match convention`);
+  }
+};
+
+const validateProcessingType = (filepath, errors, processingType) => {
+  if (processingType === undefined) {
+    errors.push(`processingType not set for ${filepath}`);
+    return;
+  }
+
+  if (
+    processingType !== "uplink_decoder" &&
```

Comment:
Technically, these value could also be all capitalised. E.g. `DOWNLINK_ENCODER`

Although, I think it would be nice to have them all lower-case in the device-type-library, the backend would allow for it and, thus, we might want to allow it here too?

Reply:
@janickakenza we should agree on one standard, both options are fine for me

Reply:
@mariusgiger I'd argue to go with snake-case, as we do not have use capitalisation for the other fields which could be interpreted as enums (e.g. `type`). what's your opinion?